.gitlab/source_test/golang_deps_generate.yml

```diff
@@ -0,0 +1,22 @@
+---
+# golang_deps_generate stage
+# Contains the step to build the golang dependency chain and preserve it
+golang_deps_generate:
+  rules:
+    !reference [.on_main_or_release_branch]
+  stage: source_test
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["runner:main"]
+  needs: ["linux_x64_go_deps"]
+  variables:
+    ARCH: "x64"
+  before_script:
+    - !reference [.retrieve_linux_go_deps]
+    - python3 -m pip install --upgrade --ignore-installed pip setuptools
+    - python3 -m pip install -r requirements.txt
+  script:
+    - inv agent.build-dep-tree
+    - tar -czf dependency_tree.tgz dependency_tree.txt
+  artifacts:
```

Comment:
We should set explicitly the `expires_in` property here. Do we want to keep them forever?

Reply:
I think we actually may want to keep them around forever (or at least a very long time). I don't know if we can be sure that we won't need an idea about how an artifact was made (especially in regards to the fed sector requirements around SBOMs)